hpx/lcos/latch.hpp

```diff
@@ -0,0 +1,143 @@
+//  Copyright (c) 2007-2015 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_LCOS_LATCH_APR_19_2015_1002AM)
+#define HPX_LCOS_LATCH_APR_19_2015_1002AM
+
+#include <hpx/exception.hpp>
+#include <hpx/include/client.hpp>
+#include <hpx/lcos/server/latch.hpp>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace lcos
+{
+    class latch : public components::client_base<latch, lcos::server::latch>
+    {
+        typedef components::client_base<latch, lcos::server::latch> base_type;
+
+    public:
+        latch()
+        {}
+
+        /// Initialize the latch
+        ///
+        /// Requires: count >= 0.
+        /// Synchronization: None
+        /// Postconditions: counter_ == count.
+        ///
+        explicit latch(std::ptrdiff_t count)
```

Comment:
Using a signed type makes it possible to detect and react to precondition violations, instead of wrapping around and pretending all is good. Unsigned is not that much about non-negative as it is about modulo arithmetics. That could go either way as far as I care, what I would like to see is this argument actually being used (or a small comment explaining why this is needed when it just behaves as a default constructor)